LeaderboardBackend/Controllers/Requests/Register.cs

```diff
@@ -1,9 +1,14 @@
+using System.ComponentModel.DataAnnotations;
 namespace LeaderboardBackend.Controllers.Requests;
 
-public record RegisterRequest
+public record RegisterRequest : LoginRequest
 {
+	[Required]
+	[RegularExpression("/([a-zA-Z][-_']?){1,12}[a-zA-Z]/",
+		ErrorMessage = "Your name must be between 2 and 25 characters, made up of letters sandwiching zero or one hyphen, underscore, or apostrophe.")]
 	public string Username { get; set; } = null!;
-	public string Email { get; set; } = null!;
-	public string Password { get; set; } = null!;
+
+	[Required]
+	[Compare("Password", ErrorMessage = "This must match your password.")]
```

Comment:
I think this should say "The password confirmation must match you password" instead of "this".